pkg/dependencies/factory.go

```
@@ -637,6 +638,9 @@ func (f *Factory) WithClusterManager(clusterConfig *v1alpha1.Cluster) *Factory {
 		if f.dependencies.ClusterManager != nil {
 			return nil
 		}
+		if f.dependencies.CliConfig == nil {
+			return nil
```

can we return an error here? or just use `0`